DFC.Digital/DFC.Digital.Web.Sitefinity.DfcSearchModule/SearchIndexEnhancers/JobProfileIndexEnhancer.cs

```diff
@@ -46,18 +47,21 @@ public JobProfileIndex GetRelatedFieldsWithUrl(JobProfileIndex jobProfileIndex)
             return jobProfileIndex;
         }
 
-        public async Task<string> GetSalaryRangeAsync()
+        public async Task<JobProfileIndex> GetSalaryRangeAsync(JobProfileIndex jobProfileIndex)
         {
-            if (JobProfile.IsLMISalaryFeedOverriden.HasValue && JobProfile.IsLMISalaryFeedOverriden.Value != true)
+            if (JobProfile.IsLMISalaryFeedOverriden.HasValue && JobProfile.IsLMISalaryFeedOverriden.Value != true && !string.IsNullOrWhiteSpace(JobProfile.SOCCode))
             {
                 var salary = await salaryService.GetSalaryBySocAsync(JobProfile.SOCCode);
-                JobProfile.SalaryStarter = salaryCalculator.GetStarterSalary(salary);
-                JobProfile.SalaryExperienced = salaryCalculator.GetExperiencedSalary(salary);
+                jobProfileIndex.SalaryStarter = Convert.ToDouble(salaryCalculator.GetStarterSalary(salary));
+                jobProfileIndex.SalaryExperienced = Convert.ToDouble(salaryCalculator.GetExperiencedSalary(salary));
+            }
+            else
+            {
+                jobProfileIndex.SalaryStarter = Convert.ToDouble(JobProfile.SalaryStarter);
+                jobProfileIndex.SalaryExperienced = Convert.ToDouble(JobProfile.SalaryExperienced);
```

Comment:
do we need additional casting? isnt it possible to change the method signature or the model to match?

Reply:
Sitefinity defaults to decimal and Azure search requires double hence the conversion